diagnose-api-breaking-changes does not understand submodules

### Is it reproducible with SwiftPM command-line tools: `swift build`, `swift test`, `swift package` etc?

- [X] Confirmed reproduction steps with SwiftPM CLI. The description text _must_ include reproduction steps with either of command-line SwiftPM commands, `swift build`, `swift test`, `swift package` etc.

### Description

When running diagnose-api-breaking-changes against a package which contains git submodules the tool does not appear to understand that it needs to get the submodules.

The project in question can be found here - https://github.com/apple/swift-async-dns-resolver

### Expected behavior

API breakage tool should checkout submodules as required.

### Actual behavior

It appears not too...

swift package diagnose-api-breaking-changes main
Building for debugging...
[71/71] Compiling AsyncDNSResolver DNSResolver_c-ares.swift
Build complete! (2.01s)
warning: '92ae16ff641eec3a6bb49bec723870ce5a02594a-checkout': found 1 file(s) which are unhandled; explicitly declare them as resources or exclude from the target
    /Users/petera/XCode/swift-async-dns-resolver/.build/arm64-apple-macosx/apidiff/92ae16ff641eec3a6bb49bec723870ce5a02594a-checkout/Sources/AsyncDNSResolver/Docs.docc
Building for debugging...
[73/73] Compiling AsyncDNSResolver AsyncDNSResolver.swift
Build complete! (16.02s)
error: baseline for AsyncDNSResolver contains no symbols, swift-api-digester output: <module-includes>:1:9: note: in file included from <module-includes>:1:
#import "/Users/petera/XCode/swift-async-dns-resolver/.build/arm64-apple-macosx/apidiff/92ae16ff641eec3a6bb49bec723870ce5a02594a-checkout/Sources/CAsyncDNSResolver/include/CAsyncDNSResolver.h"
        ^
/Users/petera/XCode/swift-async-dns-resolver/.build/arm64-apple-macosx/apidiff/92ae16ff641eec3a6bb49bec723870ce5a02594a-checkout/Sources/CAsyncDNSResolver/include/CAsyncDNSResolver.h:23:10: note: in file included from /Users/petera/XCode/swift-async-dns-resolver/.build/arm64-apple-macosx/apidiff/92ae16ff641eec3a6bb49bec723870ce5a02594a-checkout/Sources/CAsyncDNSResolver/include/CAsyncDNSResolver.h:23:
#include "../c-ares/include/ares.h"
         ^
/Users/petera/XCode/swift-async-dns-resolver/.build/arm64-apple-macosx/apidiff/92ae16ff641eec3a6bb49bec723870ce5a02594a-checkout/Sources/CAsyncDNSResolver/include/../c-ares/include/ares.h:22:10: error: 'ares_build.h' file not found
#include "ares_build.h"    /* c-ares build definitions */
         ^
<unknown>:0: error: could not build Objective-C module 'CAsyncDNSResolver'
Failed to load module: AsyncDNSResolver
error: fatalError


### Steps to reproduce

1. Check out https://github.com/apple/swift-async-dns-resolver (and submodules)
2. Make a change.
3. Try and see if you've changed the API - swift package diagnose-api-breaking-changes main

### Swift Package Manager version/commit hash

6.0

### Swift & OS version (output of `swift --version ; uname -a`)

Swift 6.0 - darwin and linux both seem to not work.

--- a/Fixtures/Miscellaneous/APIDiff/CIncludePath/Package.swift
+++ b/Fixtures/Miscellaneous/APIDiff/CIncludePath/Package.swift
@@ -1,0 +1,27 @@
+// swift-tools-version: 6.0
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+
+let package = Package(
+    name: "Sample",
+    products: [
+        .library(
+            name: "Sample",
+            targets: ["Sample"]
+        ),
+    ],
+    targets: [
+        .target(
+            name: "CSample",
+            sources: ["./vendorsrc/src"],
+            cSettings: [
+                .headerSearchPath("./vendorsrc/include"),
+            ]
+        ),
+        .target(
+            name: "Sample",
+            dependencies: ["CSample"]
+        ),
+    ]
+)

--- a/Fixtures/Miscellaneous/APIDiff/CIncludePath/Sources/CSample/include/CSample.h
+++ b/Fixtures/Miscellaneous/APIDiff/CIncludePath/Sources/CSample/include/CSample.h
@@ -1,0 +1,3 @@
+
+#include "config.h"
+#include "../vendorsrc/include/vendor.h"

--- a/Fixtures/Miscellaneous/APIDiff/CIncludePath/Sources/CSample/include/config.h
+++ b/Fixtures/Miscellaneous/APIDiff/CIncludePath/Sources/CSample/include/config.h
@@ -1,0 +1,1 @@
+#define HAVE_VENDOR_CONFIG

--- a/Fixtures/Miscellaneous/APIDiff/CIncludePath/Sources/CSample/vendorsrc/include/vendor.h
+++ b/Fixtures/Miscellaneous/APIDiff/CIncludePath/Sources/CSample/vendorsrc/include/vendor.h
@@ -1,0 +1,2 @@
+
+#include "config.h"

--- a/Fixtures/Miscellaneous/APIDiff/CIncludePath/Sources/CSample/vendorsrc/src/vendor.c
+++ b/Fixtures/Miscellaneous/APIDiff/CIncludePath/Sources/CSample/vendorsrc/src/vendor.c
@@ -1,0 +1,6 @@
+#include "vendor.h"
+
+int vendor__func(int n)
+{
+  return 0;
+}

--- a/Fixtures/Miscellaneous/APIDiff/CIncludePath/Sources/Sample/Sample.swift
+++ b/Fixtures/Miscellaneous/APIDiff/CIncludePath/Sources/Sample/Sample.swift
@@ -1,0 +1,1 @@
+import CSample

--- a/Sources/Build/BuildPlan/BuildPlan.swift
+++ b/Sources/Build/BuildPlan/BuildPlan.swift
@@ -593,6 +593,7 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
                 if let includeDir = targetDescription.moduleMap?.parentDirectory {
                     arguments += ["-I", includeDir.pathString]
                 }
+                arguments += ["-I", targetDescription.clangTarget.includeDir.pathString]
             }
         }
 

--- a/Tests/CommandsTests/APIDiffTests.swift
+++ b/Tests/CommandsTests/APIDiffTests.swift
@@ -263,6 +263,16 @@ final class APIDiffTests: CommandsTestCase {
         }
     }
 
+    func testAPIDiffOfVendoredCDependency() async throws {
+        try skipIfApiDigesterUnsupportedOrUnset()
+        try await fixture(name: "Miscellaneous/APIDiff/") { fixturePath in
+            let packageRoot = fixturePath.appending("CIncludePath")
+            let (output, _) = try await execute(["diagnose-api-breaking-changes", "main"], packagePath: packageRoot)
+
+            XCTAssertMatch(output, .contains("No breaking changes detected in Sample"))
+        }
+    }
+
     func testNoBreakingChanges() async throws {
         try skipIfApiDigesterUnsupportedOrUnset()
         try await fixture(name: "Miscellaneous/APIDiff/") { fixturePath in

